### PR TITLE
test: Fix eternal hang when waiting for VM reboot

### PR DESCRIPTION
**Pull request by martinpitt**
Commit 13f464434675 introduced a race condition into wait_reboot():
While the machine was still shutting down, wait_boot() can get past
wait_execute(), then ssh.service stops, and then it hangs eternally in
wait_user_login()'s execute() call.

Fix this by adding a 30s timeout to that execute call, and if it fails,
return back to the wait_boot() loop. Also disable connection sharing for 
this as it is prone to hangs and piling up multiple masters across
reboots.

To tell apart a timeout from some random other exception, modify Timeout
to throw a RuntimeError instead of a generic Exception (with Python 3
this can become a proper TimeoutError).

### CODE REVIEW

**Comment by martinpitt** [comment]
Getting some `Failure: Timeout waiting for system to shut down`, this needs more thought.

**Comment by martinpitt** [comment]
I found a more robust approach for this which avoids the new race condition introduced in the previous version. I locally tested this with `check-kdump` (PR #5886) and `check-packagekit` and `check-shutdown-restart` (the two which regressed before).

File: test/common/testvm.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def handle_timeout(self, signum, frame):
                 self.machine.ssh_process.terminate()
             self.machine.disconnect()
 
-        raise Exception(self.error_message)
+        raise RuntimeError(self.error_message)
```
Comment (stefwalter) [inline test/common/testvm.py]: Validated this change:

```
>>> x = RuntimeError("test")
>>> x
RuntimeError('test',)
>>> isinstance(x, Exception)
True
```

File: test/common/testvm.py [hunk 2]
```diff
@@ -212,30 +212,34 @@ def wait_user_login(self):
            user sessions are allowed (and cockit-ws will let "admin"
            in) before declaring a test machine as "booted".
 
-           Returns the boot id of the system.
+           Returns the boot id of the system, or None if ssh timed out.
```
Comment (stefwalter) [inline test/common/testvm.py]: There are no other callers of wait_user_login() so this makes sense.

File: test/common/testvm.py (hunk 1 above)
Comment (martinpitt) [inline test/common/testvm.py]: Right, `Exception` is the mother of all evilW^W^W^W^base class of all exceptions ☺

**Pull request closed without merging**